pinot-core/src/main/java/com/linkedin/pinot/core/data/partition/PartitionFunctionFactory.java

```diff
@@ -27,7 +27,8 @@
 
   // Enum for various partition functions to be added.
   public enum PartitionFunctionType {
-    MODULO
+    modulo,
+    defaultkafkapartitioner
```

Comment:
I wanted to support case insensitive naming, ie any casing of name can be specified in config. Matching enum with any kind of casing on string is easier with this approach.